static/app/components/modals/inviteMembersModal/inviteMembersFooter.tsx

```diff
@@ -6,16 +6,34 @@ import InviteButton from 'sentry/components/modals/inviteMembersModal/inviteButt
 import {InviteMembersContext} from 'sentry/components/modals/inviteMembersModal/inviteMembersContext';
 import InviteStatusMessage from 'sentry/components/modals/inviteMembersModal/inviteStatusMessage';
 import {space} from 'sentry/styles/space';
+import useOrganization from 'sentry/utils/useOrganization';
 
 interface Props {
   canSend: boolean;
 }
 
 export default function InviteMembersFooter({canSend}: Props) {
-  const {complete, inviteStatus, invites, sendInvites, sendingInvites, willInvite} =
-    useContext(InviteMembersContext);
+  const organization = useOrganization();
+  const {
+    complete,
+    inviteStatus,
+    invites,
+    pendingInvites,
+    sendInvites,
+    sendingInvites,
+    willInvite,
+  } = useContext(InviteMembersContext);
   const isValidInvites = invites.length > 0;
 
+  const removeSentInvites = () => {
+    const statuses = Object.keys(inviteStatus);
+    statuses.forEach(status => {
+      if (!pendingInvites.emails.has(status)) {
+        delete inviteStatus[status];
+      }
+    });
+  };
```

Comment:
We shouldn't be directly mutating context state as it wont dispatch react updates and could lead to weird behavior down the line as we are relying on some external rerender to update this.